Error when building some C++ source

When building with `resolver: lts-10.1` in `stack.yaml`:

```
  [53 of 71] Compiling OpenCV.ImgProc.StructuralAnalysis ( .stack-work/dist/x86_64-linux/Cabal-2.0.1.0/build/OpenCV/ImgProc/StructuralAnalysis.hs, .stack-work/dist/x86_64-linux/Cabal-2.0.1.0/build/OpenCV/ImgProc/StructuralAnalysis.o )
    /tmp/ghc18962_0/ghc_578.cpp: In function ‘void inline_c_OpenCV_ImgProc_StructuralAnalysis_2(cv::Mat*, int32_t, int32_t, int32_t*, cv::Point2i****, cv::Vec4i***, int32_t**)’:
    
    /tmp/stack4848/opencv-0.0.2.0//tmp/ghc18962_0/ghc_578.cpp:59:40: error:
         error: ‘>>’ should be ‘> >’ within a nested template argument list
               std::vector<std::vector<cv::Point>> contours;
                                                ^
       |
    59 |       std::vector<std::vector<cv::Point>> contours;
       |                                        ^
    
    /tmp/stack4848/opencv-0.0.2.0//tmp/ghc18962_0/ghc_578.cpp:83:45: error:
         error: ‘>>’ should be ‘> >’ within a nested template argument list
               for (std::vector<std::vector<cv::Point>>::size_type i = 0; i < contours.size(); i++) {
                                                     ^
       |
    83 |       for (std::vector<std::vector<cv::Point>>::size_type i = 0; i < contours.size(); i++) {
       |                                             ^
    `gcc' failed in phase `C Compiler'. (Exit code: 1)
```

It's fine with `lts-9.10` (as used in your `stack.yaml`), but thought you might want to know.

Not yet @basvandijk. I tried reproducing the build using a Stack's [Docker integration](https://docs.haskellstack.org/en/stable/docker_integration/) (I switch between Mac and Ubuntu, so I'd like to have something consistent), but with that I get:

```
 [57 of 71] Compiling OpenCV.ImgProc.GeometricImgTransform ( .stack-work/dist/x86_64-linux-dk1d9d6f18b8ece84b04ed64c4bc333808/Cabal-2.0.1.0/build/OpenCV/ImgProc/GeometricImgTransform.hs, .stack-work/dist/x86_64-linux-dk1d9d6f18b8ece84b04ed64c4bc333808/Cabal-2.0.1.0/build/OpenCV/ImgProc/GeometricImgTransform.o )
    /tmp/ghc181_0/ghc_654.cpp: In function ‘cv::Exception* inline_c_OpenCV_ImgProc_GeometricImgTransform_6(cv::Mat*, cv::Mat*, cv::Mat*, int32_t, int32_t, cv::Scalar*)’:

    /Users/dtapley/dev/haskell-opencv/opencv//tmp/ghc181_0/ghc_654.cpp:140:15: error:
         warning: extended initializer lists only available with -std=c++11 or -std=gnu++11
                     , {}
                       ^
        |
    140 |             , {}
        |               ^
```

Have you looked in to using Stack with Docker?
I may open a new issue / PR to do that.

--- a/opencv-extra/opencv-extra.cabal
+++ b/opencv-extra/opencv-extra.cabal
@@ -74,7 +74,7 @@ library
         , inline-c-cpp >= 0.2.1
 
     cc-options: -Wall -std=c++11
-    ghc-options: -Wall -fwarn-incomplete-patterns -funbox-strict-fields
+    ghc-options: -Wall -fwarn-incomplete-patterns -funbox-strict-fields -optc-std=c++11
     if flag(internal-documentation)
         cpp-options: -DENABLE_INTERNAL_DOCUMENTATION
 

--- a/opencv/opencv.cabal
+++ b/opencv/opencv.cabal
@@ -106,7 +106,7 @@ library
         , inline-c-cpp >= 0.2.1
 
     cc-options: -Wall -std=c++11
-    ghc-options: -Wall -fwarn-incomplete-patterns -funbox-strict-fields
+    ghc-options: -Wall -fwarn-incomplete-patterns -funbox-strict-fields -optc-std=c++11
     if flag(internal-documentation)
         cpp-options: -DENABLE_INTERNAL_DOCUMENTATION
 

